src/vm/amd64/crthelpers.S

```diff
@@ -13,27 +13,61 @@
 // implementation.
 //
 
+// void JIT_MemSet(void* dest, int c, size_t count)
+// 
+// Purpose:
+//    Sets the first "count" bytes of the block of memory pointed byte
+//    "dest" to the specified value (interpreted as an unsigned char).
+// 
+// Entry:
+//    RDI: void* dest    - Pointer to the block of memory to fill.
+//    RSI: int c         - Value to be set.
+//    RDX: size_t count  - Number of bytes to be set to the value.
+// 
+// Exit:
+// 
+// Uses:
+// 
+// Exceptions:
+// 
 LEAF_ENTRY JIT_MemSet, _TEXT
-        test rdx, rdx
-        jz   Exit_MemSet
+        test    rdx, rdx                // check if count is zero
+        jz      Exit_MemSet             // if zero, no bytes to set
 
-        cmp  byte ptr [rdi], 0
+        cmp     byte ptr [rdi], 0       // check dest for null
 
-        jmp  C_PLTFUNC(memset)
+        jmp     C_PLTFUNC(memset)       // forward to the CRT implementation
 
 Exit_MemSet:
         ret
 
 LEAF_END_MARKED JIT_MemSet, _TEXT
 
+// void JIT_MemCpy(void* dest, const void* src, size_t count)
+//
+// Purpose:
+//    Copies the values of "count" bytes from the location pointed to
+//    by "src" to the memory block pointed by "dest".
+//
+// Entry:
+//    RDI: void* dest             - Pointer to the destination array where content is to be copied.
+//    RSI: const void* src        - Pointer to the source of the data to be copied.
+//    RDX:  size_t count           - Number of bytes to copy.
+//
+// Exit:
+//
+// Uses:
+//
+// Exceptions:
+//
 LEAF_ENTRY JIT_MemCpy, _TEXT
-        test rdx, rdx
-        jz   Exit_MemCpy
+        test    rdx, rdx                // check if count is zero
+        jz      Exit_MemCpy             // if zero, no bytes to set
 
-        cmp  byte ptr [rdi], 0
-        cmp  byte ptr [rsi], 0
+        cmp     byte ptr [rdi], 0       // check dest for null
+        cmp     byte ptr [rsi], 0       // check src for null
 
-        jmp  C_PLTFUNC(memcpy)
+        jmp     C_PLTFUNC(memmove)      // forward to the CRT implementation, using memmove to handle overlapping buffers
```

Comment:
This was the only code change to the Unix version. The Windows version was very explicit about needing to handle overlapping buffers, and so `memmove` is the appropriate thing to forward to.

I would assume the same conditions apply to Unix.

Reply:
> The Windows version was very explicit about needing to handle overlapping buffers

That was for better compatibility with .NET Framework. It is not required by the ECMA 335: The behavior of cpblk is unspecified if the source and destination areas overlap.`

memmove is a bit slower. I do not think we need to be making the JIT helper slower on Unix...

Reply:
Do we still need it for compat on Windows? I would think consistency here is also important.

(I'll keep `memmove` on Windows and `memcpy` on Unix for now, but I think it would  be a good discussion to have).

Reply:
If nothing else, it might be good to have `JIT_MemCpy` and `JIT_MemMove`, and then the JIT can more readily decide which is appropriate based on the scenario. There may be scenarios where overlapping is desirable and several where it is not needed. And, it appears as though more than just `cpblk` goes through these.

Reply:
IIRC, memcpy as alias for memmove in Windows x64 CRT. So even if we have changed Windows to call memcpy, the unspecified behavior is still going to be different between Windows and Unix.